default/methods/source_control/file/file_browser.py

```diff
@@ -575,55 +612,58 @@ def __build_media_type_value(self) -> str:
             media_type_query = 'audio'
         return media_type_query
 
-    def file_view_core(
-        self,
-        mode = "serialize"):
-        """
-        mode
-            serialize is in context of web, ie serialize the resulting list
-                currently defaults to this context
-            objects returns the database objects, ie for auto commit
-
-        """
-        output_file_list = []
-        limit_counter = 0
-        file_count = 0  # File count includes ones we don't actually query
-        # outside of limits...
-
-        if self.metadata['file_view_mode'] is None or \
-            self.metadata['file_view_mode'] not in ["changes", "annotation", "home", "task", "explorer", "base", 'ids_only']:
-            self.log['error']['file_view_mode'] = 'Invalid file_view_mode "{}"'.format(self.metadata['file_view_mode'])
-            return None
-        ignore_id_list = None
+    def get_query_file_list(self):
+        working_dir_file_list, count = self.build_and_execute_query(
+            limit = self.metadata["limit"],
+            offset = self.metadata["start_index"],
+        )
+        if not working_dir_file_list or regular_log.log_has_error(self.log):
+            return None, None
 
-        # For creating / viewing Jobs
+        return working_dir_file_list, count
+    
+    def get_job_file_list(self):
```

Comment:
`get_job_file_list` replaces [this block](https://github.com/diffgram/diffgram/blob/38d12c685c47a53061bcbe9b80c1ae257e651a84/default/methods/source_control/file/file_browser.py#L602-L625) of code